### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by davidgs**
This is a fairly major refactor.
1. I moved things to separate files for make things easier and more readable. 
2. I fixed all PATH references to be more robust
3. Fixed the problem where `pio` may not actually be in the path.

### CODE REVIEW

File: krate.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+/**
+ * Initialize the Krate struct
+ */
+func (k *Krate) Initialize() error {
+	user, err := user.Current()
+	if err != nil {
+		return err
+	}
+	k.username = user.Username     // get the current user
+	if runtime.GOOS != "windows" { // Windows Path
```
Comment (audreylim) [inline krate.go]: I think you meant // Non-windows path here in the comment.

File: krate.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+ */
+func (k *Krate) Help() {
+	fmt.Println("Usage: krate [options...] <arg>")
+	fmt.Println(" help                    List all available commands")
+	fmt.Println(" init                    Initializes project with basic libraries and code")
+	fmt.Println(" deps install            Install dependencies specified in platformio.ini file")
+	fmt.Println(" build                   Compiles project")
+	fmt.Println(" send <ip address>       Send firmware OTA to kublet by specifiying ip address")
+	fmt.Println(" monitor                 Monitors logs")
+	fmt.Println(" publish									Creates manifest file based on prompts and user inputs for publishing on Kublet community")
```
Comment (audreylim) [inline krate.go:135]: Extra space here.

File: krate.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	kublet/KGFX@^0.0.14
+	kublet/OTAServer@^1.0.4
+monitor_speed = 460800
+`
+
+/**
+ * Initialize the Krate struct
+ */
+func (k *Krate) Initialize() error {
+	user, err := user.Current()
```
Comment (audreylim) [inline krate.go:109]: Nice.

File: krate.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+	}
+}
+
+
+func (k *Krate) create(p string) (*os.File, error) {
+	if err := os.MkdirAll(filepath.Dir(p), 0770); err != nil {
+		return nil, err
+	}
+	return os.Create(p)
+}
```
Comment (audreylim) [inline krate.go:223]: Nit: extra line needed.

**Review by audreylim — APPROVED** [review]
Looks great, just a few comments.

File: krate.go (hunk 3 above)
Comment (davidgs) [reply krate.go:109]: Thanks! :-)

File: krate.go (hunk 1 above)
Comment (davidgs) [reply krate.go]: I did. And I meant Windows 2 lines down. Backwards!

File: krate.go (hunk 4 above)
Comment (davidgs) [reply krate.go:223]: And comments for each function. Oops!

**Comment by davidgs** [comment]
Should be good to go now.

File: krate.go [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+ */
+func (k *Krate) Help() {
+	fmt.Println("Usage: krate [options...] <arg>")
+	fmt.Println(" help                    List all available commands")
+	fmt.Println(" init                    Initializes project with basic libraries and code")
+	fmt.Println(" deps install            Install dependencies specified in platformio.ini file")
+	fmt.Println(" build                   Compiles project")
+	fmt.Println(" send <ip address>       Send firmware OTA to kublet by specifiying ip address")
+	fmt.Println(" monitor                 Monitors logs")
+	fmt.Println(" publish									Creates manifest file based on prompts and user inputs for publishing on Kublet community")
```
Comment (audreylim) [inline krate.go:135]: Not sure why this is still showing up as empty space. Let's merge it first and I'll fix it.

**Review by audreylim — COMMENTED** [review]
LGTM, let's merge it.

File: krate.go (hunk 5 above)
Comment (audreylim) [reply krate.go:135]: Probably a GitHub issue, resolved.

**Pull request merged**